packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -70,11 +123,150 @@ - (void)handleProductRequestMethodCall:(FlutterMethodCall *)call result:(Flutter
                                  details:call.arguments]);
       return;
     }
+    for (SKProduct *product in response.products) {
+      [self.productsCache setObject:product forKey:product.productIdentifier];
+    }
     result([FIAObjectTranslator getMapFromSKProductsResponse:response]);
     [weakSelf.requestHandlers removeObject:handler];
   }];
 }
 
+- (void)createPaymentWithProductID:(FlutterMethodCall *)call result:(FlutterResult)result {
+  if (![call.arguments isKindOfClass:[NSString class]]) {
+    result([FlutterError
+        errorWithCode:@"storekit_invalide_argument"
+              message:@"Argument type of createPaymentWithProductID is not a string."
+              details:call.arguments]);
+    return;
+  }
+  NSString *productID = call.arguments;
+  SKProduct *product = [self.productsCache objectForKey:productID];
+  if (!product) {
+    result([FlutterError
+        errorWithCode:@"storekit_product_not_found"
+              message:@"Cannot find the product. To create a payment of a product, you must query "
+                      @"the product with SKProductRequestMaker.startProductRequest."
+              details:call.arguments]);
+    return;
+  }
+  SKPayment *payment = [SKPayment paymentWithProduct:product];
+  [self.paymentsCache setObject:payment forKey:productID];
+  result([FIAObjectTranslator getMapFromSKPayment:payment]);
+}
+
+- (void)addPayment:(FlutterMethodCall *)call result:(FlutterResult)result {
+  if (![call.arguments isKindOfClass:[NSDictionary class]]) {
+    result([FlutterError errorWithCode:@"storekit_invalide_argument"
+                               message:@"Argument type of addPayment is not a map"
+                               details:call.arguments]);
+    return;
+  }
+  NSDictionary *paymentMap = (NSDictionary *)call.arguments;
+  NSString *productID = [paymentMap objectForKey:@"productID"];
+  SKPayment *payment = [self.paymentsCache objectForKey:productID];
+  // User can use  payment object with mutable = true and add simulatesAskToBuyInSandBox = true to
+  // test the payment flow.
+  if (!payment || [paymentMap[@"mutable"] boolValue] == YES) {
+    SKMutablePayment *mutablePayment = [[SKMutablePayment alloc] init];
+    mutablePayment.productIdentifier = productID;
+    NSNumber *quantity = [paymentMap objectForKey:@"quantity"];
+    if (quantity) {
+      mutablePayment.quantity = quantity.integerValue;
+    }
+    NSString *applicationUsername = [paymentMap objectForKey:@"applicationUsername"];
+    mutablePayment.applicationUsername = applicationUsername;
+    if (@available(iOS 8.3, *)) {
+      mutablePayment.simulatesAskToBuyInSandbox =
+          [[paymentMap objectForKey:@"simulatesAskToBuyInSandBox"] boolValue];
+    }
+    [self.paymentQueueHandler addPayment:payment];
```

Comment:
nit, subjective: You can git rid of this line and the else case here by always executing this statement at the bottom of the function, outside of the if block entirely.

Reply:
it is actually a bug, it should be 
```
    [self.paymentQueueHandler addPayment:mutablePayment];
```